### PR TITLE
Fixes #32980 - Add className and headProps to PageLayout

### PR DESCRIPTION
**Pull request by Ron-Lavi**
Used in #8596

### CODE REVIEW

**Review by amirfefer — COMMENTED** [review]
Thanks @laviro, 

Shouldn't #8596 be based on PF4 components?
This `PageLayout` is PF3 based.

**Comment by Ron-Lavi** [comment]
> Thanks @laviro,
> 
> Shouldn't #8596 be based on PF4 components?
> This `PageLayout` is PF3 based.

IMHO we can reuse it and upgrade its layout overtime to PF4 without breaking current usages

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js [hunk 1]
```diff
@@ -20,11 +20,13 @@ const PageLayout = ({
...
   children,
+  className,
+  title,
 }) => (
-  <div id="main">
+  <div id="main" className={className}>
     <div id="react-content">
       <Head>
-        <title>{header}</title>
+        <title>{title || header}</title>
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: Why do we need both `title` and `header` props?

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js [hunk 2]
```diff
@@ -115,6 +119,8 @@ PageLayout.propTypes = {
 };
 
 PageLayout.defaultProps = {
+  className: '',
```
Comment (tbrisker) [inline webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: Should this default to `undefined` so we don't get an empty prop when it isn't used?

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js (hunk 1 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: atm, the default title is the header of the page,
we should allow an option to edit the title as some plugins already do,
so for example the title would be: `My App - Architectures`, and the page header can remain `Architectures`

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js (hunk 2 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: I think that passing `undefined` will still result in `<div class />` but didn't check,
We can do something like:
```js
const cxProps = className ? {} : { className };
// later in div
<div {...cxProps} />
```
but I wonder if it really worth the effort?

Comment (tbrisker) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: Actually I think react removes `undefined` and `null` attributes:
https://github.com/facebook/react/blob/8b88ac2592c5f555f315f9440cbb665dd1e7457a/packages/react-dom/src/shared/DOMProperty.js#L143-L152

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js (hunk 1 above)
Comment (tbrisker) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: But wouldn't then these plugins need to also know the header (e.g. `Architectures` in your example) ? Would it be better in such a case to use `titleTemplate` attribute in Helmet?

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js (hunk 2 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: Awesome, will change to null then!

File: webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js (hunk 1 above)
Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: Agreed, thanks @tbrisker

Comment (Ron-Lavi) [reply webpack/assets/javascripts/react_app/routes/common/PageLayout/PageLayout.js]: passing now `headProps` into the `<Head />` component to cover more options

**Comment by MariaAga** [comment]
@laviro where does #8596 use `headProps`?

**Comment by Ron-Lavi** [comment]
closing for now, will open if this will be still needed, thanks all!

**Pull request closed without merging**